### PR TITLE
ZSTD: Fixing warning about large integer implicitly truncated to unsigned t…

### PR DESCRIPTION
**Pull request by oshadura**
…ype in ZipZSTD.cxx

### CODE REVIEW

File: core/zstd/src/ZipZSTD.cxx [hunk 1]
```diff
@@ -20,7 +20,7 @@
 
 static const int kHeaderSize = 9;
 
-static const size_t errorCodeSmallBuffer = 18446744073709551546U;
+static const unsigned long long errorCodeSmallBuffer = 18446744073709551546U;
```
Comment (Axel-Naumann) [inline core/zstd/src/ZipZSTD.cxx]:
```suggestion
static const size_t errorCodeSmallBuffer = (size_t)-70;
```

File: core/zstd/src/ZipZSTD.cxx [hunk 2]
```diff
@@ -35,9 +35,11 @@ void R__zipZSTD(int cxlevel, int *srcsize, char *src, int *tgtsize, char *tgt, i
                                         2*cxlevel);
 
     if (R__unlikely(ZSTD_isError(retval))) {
-        std::cerr << "Error in zip ZSTD. Type = " << ZSTD_getErrorName(retval) <<
-        " . Code = " << retval << std::endl;
-        return;
+        if (R__unlikely(retval != static_cast<size_t>(errorCodeSmallBuffer))) {
```
Comment (Axel-Naumann) [inline core/zstd/src/ZipZSTD.cxx]:
```suggestion
        if (R__unlikely(retval != errorCodeSmallBuffer)) {
```

Comment (Axel-Naumann) [reply core/zstd/src/ZipZSTD.cxx]: Actually:
```
if (R__unlikely(ZSTD_isError(retval))) {
  if (R__unlikely(retval != static_cast<size_t>(errorCodeSmallBuffer))) {
```
should be combined as
```
if (R__unlikely(ZSTD_isError(retval) && retval != static_cast<size_t>(errorCodeSmallBuffer))) {
```
(i.e. have `R__unlikely` surround the whole condition)

**Comment by oshadura** [comment]
Failure on Ubuntu is caused by unrelated warning.

**Pull request merged**